feature-utils/poly-look/src/react-components/inputs/dropdown.jsx

```diff
@@ -0,0 +1,120 @@
+import React, { useState, useRef, useEffect } from "react";
+import { ChevronDown } from "../icons";
+
+import "./common.css";
+import "./dropdown.css";
+
+function getBodyClass(disabled, error, focused) {
+  if (disabled) return "body-disabled";
+  if ((error && focused) || focused) return "body-focused";
+  else if (error) return "body-error";
+}
+
+function filterOptions(options, text) {
+  return options.filter((option) => option.value.includes(text));
+}
+
+/** Dropdown with options filtering.
+ * @param {Object} props
+ * @param {Object} [props.initialSelection] - input value, has to contain
+ * an id and value, leave empty if the dropdown is not pre-populated
+ * @param {callback} [props.onChange] - onChange event handler
+ * @param {string} [props.name] - input name, also acts as id
+ * @param {string} [props.helperText] - input helper text
+ * @param {string} [props.label] - input label
+ * @param {number} [props.tabIndex] - input tabIndex, needed to make sure the
+ * focus/blur internal logic works
+ * @param {boolean} [props.error] - indicates if the input is in an error
+ * state or not
+ * @param {boolean} [props.disabled] - indicates if the input is disabled
+ * @param {Array[Object]} [props.options] - array containing the dropdown options,
+ * an option should have an id and a value
+ * @param {string} [props.noMatch] - text to be shown when there is no match
+ * @returns {JSX.Element}
+ */
+
+export function Dropdown({
+  initialSelection = {},
+  onChange,
+  name,
+  helperText,
+  label,
+  tabIndex,
+  error,
+  disabled,
+  options,
+  noMatch,
+}) {
+  const [open, setOpen] = useState(false);
+  const [dropdownOptions, setOptions] = useState(options);
+  const [query, setQuery] = useState(initialSelection.value || "");
+  const [selected, setSelected] = useState(initialSelection);
+  const inputRef = useRef();
+
+  let labelClass = query || open ? "label-up" : "";
+  let bodyClass = getBodyClass(disabled, error, open);
+
+  useEffect(() => {
+    onChange({ ...selected, name });
+  }, [selected]);
+
+  return (
+    <div className="poly-input-field poly-dropdown">
+      <div
+        data-testid="input-focusable"
+        className={`body ${bodyClass}`}
+        onFocus={() => {
+          setOpen(true);
+          inputRef.current.focus();
+        }}
+        onBlur={() => setOpen(false)}
+        tabIndex={tabIndex}
+      >
+        <label htmlFor={name} className={labelClass}>
+          {label}
+        </label>
+        <input
+          disabled={disabled}
```

Comment:
I don't know if you were aware of this, but you could have also done:
`<select>
        <option value={elem.value}>{elem.value}<option />
</select>`

And you wouldn't have needed to add the `ChevronDown` component and the `dropdown-list`(and I think the `setOption` neither). 

But this way you can style the icon and the dropdown-list and it looks great :)

Reply:
I wasn't sure how easy it would have been to style it so I've went with the custom implementation part.